repl/option.go

```diff
@@ -0,0 +1,84 @@
+package repl
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"strings"
+)
+
+// Option implementations can be provided to New() to configure the REPL
+// during initialization.
+type Option func(repl *REPL)
+
+// WithInput sets the REPL's input stream. `nil` defaults to bufio.Scanner
+// backed by os.Stdin
+func WithInput(in Input, errMapper func(error) error) Option {
```

Comment:
Thoughts about making an `ErrorMapper` type?  I often find this easier to use because inline function signatures get noisy quite quickly. 

I'm inclined to do something like:

```go
type ErrorMap interface{
    Map(error) error
}

type ErrorMapFunc func(error) error

func (f ErrorMapFunc) Map(err error) error {
    return f(err)
}
```

This makes it easy for users to supply complex types for `errMap`.  A motivating example:

```go
logger := log.New()  // github.com/lthibault/log/pkg

errMap := func(err error) error {
    switch {
    case errors.Is(err, io.EOF):
         logger.Warn("exiting")
        return io.EOF
    case errors.Is(err, ww.Interrupt):
        logger.Trace("keyboard interrupt")
        return nil
    }    
}

r := repl.New(runtime,
    repl.WithInput(input, repl.ErrorMapFunc(errMap))
```

See [net/http::HandlerFunc](https://golang.org/pkg/net/http/#HandlerFunc) for a similar pattern in the Go standard library.

Reply:
I agree. Will define a type for error-mapper.

Reply:
Ok fair enough.  So you're thinking `type ErrorMapper func(error) error` ?

Works for me 👍

Reply:
Yep.